update_remote_settings_records.mjs

```diff
@@ -0,0 +1,352 @@
+  })
+  .catch((e) => {
+    console.error(e);
+    return process.exit(FAILURE_RET_VALUE);
+  });
+
+async function update() {
+  console.log(`Get existing records from ${rsBrowsersCollectionEndpoint}`);
+  const records = await getRSRecords();
+  const operations = { added: [], updated: [], removed: [] };
+
+  const browsersMdn = getFlatBrowsersMdnData();
+
+  for (const browserMdn of browsersMdn) {
+    const rsRecord = records.find(
+      (record) =>
+        record.browserid == browserMdn.browserid &&
+        record.version == browserMdn.version
+    );
+    if (browserMdn.status == "retired") {
+      if (rsRecord) {
+        const succesful = await deleteRecord(rsRecord);
+        if (succesful) {
+          operations.removed.push(rsRecord);
+        }
+      }
+      continue;
+    }
+
+    if (!rsRecord) {
+      const succesful = await createRecord(browserMdn);
+      if (succesful) {
+        operations.added.push(browserMdn);
+      }
+      continue;
+    }
+
+    if (
+      rsRecord.status !== browserMdn.status ||
+      rsRecord.name !== browserMdn.name
+    ) {
+      const succesful = await updateRecord(rsRecord, browserMdn);
+      if (succesful) {
+        operations.updated.push(browserMdn);
+      }
+    }
+  }
+
+  for (const record of records) {
+    const browserMdn = browsersMdn.find(
+      (browser) =>
+        browser.browserid == record.browserid &&
+        browser.version == record.version
+    );
+    if (!browserMdn) {
+      const succesful = await deleteRecord(record);
+      if (succesful) {
+        operations.removed.push(record);
+      }
+    }
+  }
+
+  console.group("Results");
+  console.log("Added:", operations.added.length);
+  if (operations.added.length > 0) {
+    console.table(operations.added);
+  }
+  console.log("Updated:", operations.updated.length);
+  if (operations.updated.length > 0) {
+    console.table(operations.updated);
+  }
+  console.log("Removed:", operations.removed.length);
+  if (operations.removed.length > 0) {
+    console.table(operations.removed);
+  }
+  console.groupEnd();
+
+  if (
+    operations.added.length +
+      operations.updated.length +
+      operations.removed.length ==
+    0
+  ) {
+    console.log("No changes detected");
+  } else {
+    const refreshedRecords = await getRSRecords();
+    console.log("Browsers data synced ✅\nRefreshed records:");
+    console.table(refreshedRecords);
+    if (process.env.ENVIRONMENT === "dev") {
+      console.log("Approving changes");
+      await approveChanges();
+      console.log("Changes approved ✅");
+    } else {
+      console.log("Requesting review");
+      await requestReview();
+      console.log("Review requested ✅");
+    }
+  }
+}
+
+async function getRSRecords() {
+  const response = await fetch(rsBrowsersRecordsEndpoint, {
+    method: "GET",
+    headers: {
+      "Content-Type": "application/json",
+      Authorization: process.env.AUTHORIZATION,
+    },
+  });
```

Comment:
Instead of repeating these lines, you may want to do something like:

```js
async function fetchJSON(url, options = {}) {
  if (isDryRun && options.method != "GET") {
    console.log( ... );
    return;
  }
  const response = await fetch(url, {
    headers: {
      "Content-Type": "application/json",
      Authorization: process.env.AUTHORIZATION,
    },
    ...options
  });
  if (response.status >= 300) {
    ...
  }
  return await response.json();
}
```

Reply:
I feel like this would complexify the code a bit for little returned value?
What we could do to avoid repetition is having a `headers` variable that we would instantiate once so we could do 

```js
  const response = await fetch(`${rsBrowsersRecordsEndpoint}/${record.id}`, {
    method: "PUT",
    body: JSON.stringify({ data: browserMdn }),
    headers,
  });
```

This looks not too different from 

```js
  const response = await fetchJSON(`${rsBrowsersRecordsEndpoint}/${record.id}`, {
    method: "PUT",
    body: JSON.stringify({ data: browserMdn }),
  });
```

while being more explicit and flexible for each functions (checking exact status code, retrieving returned json data only when needed, having proper warning message on api call failure)

Reply:
yes, great!